app/src/main/java/org/mozilla/fenix/settings/Extensions.kt

```diff
@@ -14,6 +17,24 @@ import mozilla.components.support.ktx.android.view.putCompoundDrawablesRelative
 import org.mozilla.fenix.R
 import org.mozilla.fenix.theme.ThemeManager
 
+/**
+ * Checks for availability of network.
+ *
+ * For devices above [Build.VERSION_CODES.M] it even checks if there's internet flowing through it or not.
+ * */
+fun ConnectivityManager.isOnline(): Boolean {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        getNetworkCapabilities(activeNetwork)?.let {
+            it.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) &&
+                    it.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)
+        } ?: false
+    } else {
+        // for device below android M, there's not better way to get this.
+        // active network info can be null if there are no active networks.
+        activeNetworkInfo?.isConnected ?: false
```

Comment:
You might need to add `@Suppress("Deprecation")` here

Reply:
aah, missed this.